core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -221,6 +221,175 @@ private[spark] class BlockManager(
     new BlockManager.RemoteBlockDownloadFileManager(this)
   private val maxRemoteBlockToMem = conf.get(config.MAX_REMOTE_BLOCK_SIZE_FETCH_TO_MEM)
 
+  private abstract class BlockStoreUpdater[T](
+      blockSize: Long,
+      blockId: BlockId,
+      level: StorageLevel,
+      classTag: ClassTag[T],
+      tellMaster: Boolean,
+      keepReadLock: Boolean) {
+
+    protected def byteBuffer: ChunkedByteBuffer
```

Comment:
This abstract method is necessary for accessing the lazy val `TempFileBasedBlockStoreUpdater.bytes` (as lazy vals cannot be abstract).

Reply:
I see what you're doing but it looks a little weird. The semantics here are that this method should always return the /same/ buffer, like a `val` (at least to avoid creating unnecessary buffers).

So it needs at least a comment explaining that. But that makes me think that the abstraction here is a little funny and this could be solved some other way... can't really suggest something off the top of my head, though.